doc/netplan.md

```diff
@@ -141,6 +141,48 @@ Virtual devices
 
 :    (networkd backend only) Whether to emit LLDP packets. Off by default.
 
+``receive-checksum-offload`` (bool) – since **0.104**
+
+:    (networkd backend only) If set to true, the hardware offload for
+     checksumming of ingress network packets is enabled. When unset,
+     the kernel's default will be used.
+
+``transmit-checksum-offload`` (bool) – since **0.104**
+
+:    (networkd backend only) If set to true, the hardware offload for
+     checksumming of egress network packets is enabled. When unset,
+     the kernel's default will be used.
+
+``tcp-segmentation-offload`` (bool) – since **0.104**
+
+:    (networkd backend only) If set to true, the TCP Segmentation
+     Offload (TSO) is enabled. When unset, the kernel's default will
+     be used.
+
+``tcp6-segmentation-offload`` (bool) – since **0.104**
+
+:    (networkd backend only) If set to true, the TCP6 Segmentation
+     Offload (tx-tcp6-segmentation) is enabled. When unset, the
+     kernel's default will be used.
+
+``generic-segmentation-offload`` (bool) – since **0.104**
+
+:    (networkd backend only) If set to true, the Generic Segmentation
+     Offload (GSO) is enabled. When unset, the kernel's default will
+     be used.
+
+``generic-receive-offload`` (bool) – since **0.104**
+
+:    (networkd backend only) If set to true, the Generic Receive
+     Offload (GRO) is enabled. When unset, the kernel's default will
+     be used.
+
+``large-receive-offload`` (bool) – since **0.104**
+
+:    (networkd backend only) If set to true, the Generic Receive
+     Offload (GRO) is enabled. When unset, the kernel's default will
+     be used.
+
```

Comment:
+1 on the schema here.  These are all long option names, which is not ideal, but I don't see any way to shorten them without sacrificing clarity.  (We should avoid using abbreviations; while common abbreviations exist for these options, they are much more opaque.)

Reply:
Thanks for the review @vorlonofportland . I agree that shortening the names might lead to confusing acronym-like options which is not very user friendly in the end.